topics/basics/testing_plugins/integration_tests/integration_tests_api.md

```diff
@@ -0,0 +1,197 @@
+Driver framework uses a standard Java Remote Method Invocation (RMI) protocol.
+This protocol allows accessing objects and invoking methods from tests in the JVM of the IDE.
+
+The architecture of RMI protocol is as follows:
+```mermaid
+sequenceDiagram
+    box rgb(240,240,240) Test Process
+    participant Test as Test (Client)
+    end
+
+    box rgb(240,240,240) IDE Process
+    participant IDE as IDE (Server)
+    participant Registry as RMI registry
+    end
+
+    Test->>Registry: 1. Look up remote object
+    Registry-->>Test: 2. Return stub reference
+
+    Note over Test: Stub is obtained<br/>for remote reference
+
+    activate Test
+    Test->>IDE: 3. Remote method call via stub
+    IDE->>IDE: 4. Process request and invoke actual method
+    IDE-->>Test: 5. Return serialized result
+    deactivate Test
+
+    Note over IDE,Test: Each remote call involves:<br/>- Serialization of parameters<br/>- Network transfer<br/>- Deserialization<br/>- Method invocation<br/>- Serialization of result
+```
+
+When a test needs to invoke a method on a remote object:
+
+1. The test looks up the remote object in the RMI registry (the IDE).
+2. The IDE returns a reference to the stub, which implements the same interface as the remote object.
+3. When the client calls a method on the stub:
+    * The stub serializes the method call, including any parameters.
+    * It sends the serialized data over the network to the remote server.
+    * The RMI runtime inside the IDE unmarshals the request and invokes the corresponding method on the actual remote object.
+    * The method executes, and the result is serialized and sent back to the client via the stub.
+    * The stub unmarshals the response and returns the result to the caller.
+
+Thus, the stub makes remote calls feel like local method calls, without the caller worrying about low-level networking, data conversion, or request handling.
+
+## Creating stubs
+
+To demonstrate how this works in practice, add the following code to the plugin:
+
+```kotlin
+class PluginStorage {
+  companion object {
+    @JvmStatic
+    fun getPluginStorage() = Storage("static method", listOf("static1", "static2"))
+  }
+}
+
+@Service
+class PluginService {
+  fun someMethod(): Unit = Unit
+  fun getAnswer(): Int = 42
+}
+
+@Service(Service.Level.PROJECT)
+class PluginProjectService {
+  fun getStrings(): Array<String> = arrayOf("foo", "bar")
+}
+
+data class Storage(val key: String, val attributes: List<String>)
+```
+
+This file contains one class with the static method `getPluginStorage`, and two [light services](plugin_services.md#light_services) – one application-level `PluginService` and one project-level `PluginProjectService`.
+
+The test needs to call the methods of these classes and verify their return values.
+To do this, the following stubs have to be created.
+
+```kotlin
+import com.intellij.driver.client.Remote
+
+@Remote("org.jetbrains.testPlugin.PluginStorage", plugin = "org.example.demo")
```

Comment:
While both are correct, please use `com.example.demo` for consistency. We use `com` in other places.